service/src/worker.rs

```diff
@@ -84,35 +75,38 @@ where
 		}
 		Ok(())
 	}
+}
 
-	fn peers(&self) -> WorkerResult<Vec<EnclaveMetadata>> {
-		Ok(self.node_api.all_enclaves(None)?)
+/// Looks for new peers and updates them.
+pub trait UpdatePeers {
```

Comment:
Introduced new trait that is in charge of updating the peers. Needs some thoughts when peers should be updated. Currently we update them for every block gossip..